Задача: Meta-вселенная?

piligrim-curious opened the pull request:
**Notification**: Класс оповещений;
**NotificationManager**: Обработчик оповещений;
**NotificationType**: Допустимые типы оповещений.

piligrim-curious committed:
Задача: Meta-вселенная?
Notification: Класс оповещений;
NotificationManager: Обработчик оповещений;
NotificationType: Допустимые типы оповещений.

iBerbasov commented on src/main/java/school/faang/meta/universe/NotificationManager.java:
```diff
@@ -0,0 +1,45 @@
+public class NotificationManager {
+    Map<NotificationType, Consumer<Notification>> notificationManager = new HashMap<>();
+
+    void registerHandler(NotificationType type, Consumer<Notification> handler) {
```
Нужны модификаторы доступа, эти методы должны быть public

iBerbasov commented on src/main/java/school/faang/meta/universe/NotificationManager.java:
```diff
@@ -0,0 +1,45 @@
+        System.out.println("[+] Зарегистрирован новый обработчик - " + type);
+    }
+
+    void sendNotification(Notification notification) {
```
И этот метод тоже

iBerbasov commented on src/main/java/school/faang/meta/universe/NotificationManager.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.function.Consumer;
+
+public class NotificationManager {
+    Map<NotificationType, Consumer<Notification>> notificationManager = new HashMap<>();
```
А это поле должно быть приватным, нельзя открывать вовне доступ к состоянию класса
Так же тут не совсем корректное имя - notificationManager это сам класс, в данной коллекции хранятся хендлеры , поэтому более информативным именем будет что-то вроде handlers, notificationHandlers и т.д.

piligrim-curious committed:
Задача: Meta-вселенная?
Notification: Класс оповещений;
NotificationManager: Обработчик оповещений;
NotificationType: Допустимые типы оповещений.

Изменения:
(NotificationManager.java)
* Добавлены модификаторы доступа.
* Добавлено поле final для мапы.
* Изменено название мапы.
